docs/proposals/DNSPolicy.md

```diff
@@ -0,0 +1,280 @@
+  
+
+```yaml 
+
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+name: default-policy
+spec:
+  targetRef: # defaults to gateway gvk and currrent namespace
+    name: gateway-name
+  loadBalancing:
+    weighted: # required
+     default: 10  #required, all records created get this weight
+  health:
+   ...   
+```  
+
+In order to provide GEO based DNS and allow customisation of the weighting, we need some additional information to be provided by the gateway / cluster admin about where this gateway has been placed. For example if they want to use GEO based DNS as a strategy, we need to know what GEO identifier(s) to use for each record we create and a default GEO to use as a catch all. Also if the desired load balancing approach is to provide custom weighting and no longer simply use Round Robin, we will need a way to identify which records to apply that custom weighting to based on the clusters the gateway is placed on.
+
+To solve this we will allow two new attributes to be added to the `ManagedCluster` resource as labels:
+
+```
+   kuadrant.io/lb-attribute-GEO-<GEO-code-type>: "IE"
+   kuadrant.io/lb-attribute-custom-weight: "GCP"
+```
+
+These two labels allow setting values in the DNSPolicy that will be reflected into DNS records for gateways placed on that cluster depending on the strategies used. (see the first DNSPolicy definition above to see how these values are used) or take a look at the examples at the bottom.
+
+
+example :
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+ labels:
+   kuadrant.io/lb-attribute-GEO-country-code: "IE"
+   kuadrant.io/lb-attribute-custom-weight: "GCP"
+spec:    
+```  
+
+The attributes provide the key and value we need in order to understand how to define records for a given LB address based on the DNSPolicy targeting the gateway.
+
+
+### DNS Record Structure
+
+This is an advanced topic and so is broken out into its own proposal doc [DNS Record Structure](./DNSRecordStructure.md)
+
+
+### Custom Weighting 
+
+Custom weighting will use the associated `custom-weight` attribute set on the `ManagedCluster` to decide which records should get a specific weight. The value of this attribute is up to the end user.
+
+example:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+ labels:
+   kuadrant.io/lb-attribute-custom-weight: "GCP"
```

Comment:
if a cluster has this attribute set (e.g. GCP), but that value isn't mentioned in the DNS Policy (i.e. no GCP value is specified), do we just apply the default weight to that cluster, in that scenario?

Reply:
yes